### PR TITLE
Draft: First stab at using new python library

### PR DESCRIPTION
**Pull request by DrTree**
I've had a first rough stab at using the new python library.

Initial assessment is that the Neohub API is a pain to work with!
GET_ZONES doesn't allow us to tell which is as switch or stat and is therefore borderline unusable.

This, works on my machine 😯

### CODE REVIEW

**Comment by DrTree** [comment]
Work to do:
 - Implement switches 
 - Filter out timers
 - Refactor the update loop (possibly use a coordinator)

File: custom_components/heatmiserneo/climate.py [hunk 1]
```diff
@@ -164,76 +137,55 @@ def hvac_modes(self):
...
-    # def preset_modes(self):
-    #     """Return preset modes."""
-    #     return self._preset_modes
-
-    def set_temperature(self, **kwargs):
+    async def async_set_temperature(self, **kwargs):
         """ Set new target temperature. """
-        response = self._hub.json_request({"SET_TEMP": [int(kwargs.get(ATTR_TEMPERATURE)), self._name]})
+        response = await self._hub.set_temp(
+            int(kwargs.get(ATTR_TEMPERATURE)), [self._neostat]
```
Comment (stikonas) [inline custom_components/heatmiserneo/climate.py:143]: We should probably drop `int` here. `int` will break support for setting temperature with 0.5 degree accuracy.

I should probably add some function to neohubapi that returns `target_temperature_step`, so we can round to appropriate precision... But I think simply dropping `int` would work for now.

File: custom_components/heatmiserneo/climate.py [hunk 2]
```diff
@@ -164,76 +137,55 @@ def hvac_modes(self):
...
         elif hvac_mode == HVAC_MODE_OFF:
             self._hvac_mode = HVAC_MODE_OFF
-            mode = "FROST_ON"
+            frost = True
         else:
             _LOGGER.error("Unrecognized hvac mode: %s", hvac_mode)
             return
 
-        response = self._hub.json_request({mode: [self._name]})
+        response = await self._hub.frost(frost, [self._neostat])
```
Comment (stikonas) [inline custom_components/heatmiserneo/climate.py:164]: Was there any particular reason not to use `await self._neostat.frost(frost)` ? Seems slightly shorter.

**Comment by stikonas** [comment]
As a first draft, looks quite good to me. There is still some ugliness in part where you call `get_live_data` but it's not your fault.

Maybe we can move that code to neohubapi, so that somehow library would return the NeoStat objects (ideally with support for reading its own attributes, not just for calling a few functions).

Unfortunately, this weekend I didn't have too much time to look at neohubapi...

**Comment by stikonas** [comment]
Also it might be good to setup both devices and entities. That way we can e.g. associate stuff like battery level with the devices

Might be helpful for https://github.com/MindrustUK/Heatmiser-for-home-assistant/issues/54

**Comment by stikonas** [comment]
I've tried to adjust neohubapi a bit to simplify code in this PR a bit...

**Comment by stikonas** [comment]
I've also pushed neohubapi here: https://gitlab.com/neohubapi/neohubapi

Next I need to investigate what I need to do to publish it to PyPI.

**Comment by stikonas** [comment]
Also pushed to PyPI: https://pypi.org/project/neohubapi/

**Comment by stikonas** [comment]
I'm now preparing a new revision for this PR (I think I have to create a new PR since I can't push to @DrTree branch).

It will still be WIP Draft and not ready for merge but a bit cleaned up.

**Comment by stikonas** [comment]
Shall we close this?

**Comment by DrTree** [comment]
Yes

**Pull request closed without merging**